### ISSUE TITLE
CodeLens breaks with the latest ccls. (xref)

### ISSUE THREAD

**Issue by icylogic** [MaskRay/vscode-ccls#9]
Since CodeLens has changed in the latest commit https://github.com/MaskRay/ccls/commit/8c807685ab1200b404303d2d6954a06ec176cb9c 

The result of requesting `textDocument/codeLens` changed from

![screenshot_2018-10-01_01-13-56](https://user-images.githubusercontent.com/2814357/46260392-d26bcd00-c517-11e8-8783-cec889c788a2.png)

to

![screenshot_2018-10-01_01-16-00](https://user-images.githubusercontent.com/2814357/46260394-d566bd80-c517-11e8-8173-25377fdc659e.png)

but I have no idea how to process it.

**Comment by MaskRay** [comment]
Sorry that I have squashed the commit. It is now https://github.com/MaskRay/ccls/commit/da704521b550606ad0cf3c8dcff8ad2efb54d02e

This is a **breaking change**.

What ccls did (inherited from cquery):

* If you see a code lens `1000 refs` (clickable button that encodes 1000 `Location[]`)
* The 1000 `Location[]` are encoded in the `textDocument/codeLens` response, which increases the size a lot and it also takes some time for the server to render this piece of information.

After the refactoring (the commit you've observed), 

https://github.com/maskray/ccls/blob/master/src/messages/textDocument_codeLens.cc#L40

what the server returns now is:

```javascript
"command":{
  "title":"1 var",
  "command":"ccls.xref",
  "arguments":["{\"usr\":16926293689053220079,\"kind\":2,\"field\":\"instances\"}"]}}
```

The `Location[]` has been replaced with `"arguments"`. When the user clicks the lens button, it sends a `workspace/executeCommand` request to the server and the server returns `Location[]`. This improves the performance a lot.
